src/mutex.c

```diff
@@ -69,14 +69,21 @@ void
 malloc_mutex_lock_slow(malloc_mutex_t *mutex) {
 	mutex_prof_data_t *data = &mutex->prof_data;
 
-	{//TODO: a smart spin policy
+	if (ncpus == 1) {
+		goto label_spin_done;
+	}
+
+	int cnt = 0, max_cnt = MALLOC_MUTEX_MAX_SPIN;
+	do {
+		CPU_SPINWAIT;
```

Comment:
Oh if it's a CAS loop then yes it will mess up the cache MESI thing, however the trylock implementations will always do ``` if (!is_locked) then CAS() ```. In this case if there's no change to the lock state, then there will only be a simple load to an unchanged cacheline, and therefore no cache coherent traffic generated.